controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -8,8 +16,91 @@
 
 #include <easylogging++.h>
 
+#include <algorithm>
+
 using namespace beerocks;
 
+std::string wfa_ca::wfa_ca_status(eWfaCaStatus status)
+{
+    switch (status) {
+    case RUNNING:
+        return "RUNNING";
+    case INVALID:
+        return "INVALID";
+    case ERROR:
+        return "ERROR";
+    case COMPLETE:
+        return "COMPLETE";
+    default:
+        break;
+    }
+    return std::string();
+}
+
+wfa_ca::eWfaCaCommand wfa_ca::wfa_ca_command(std::string command)
+{
+    std::transform(command.begin(), command.end(), command.begin(), ::toupper);
+
+    if (command == "CA_GET_VERSION") {
```

Comment:
A big sequence of comparisons like this is nicer as a lookup table with callback functions. In this case, it could be just a string/enum mapping. However, this function is only used within handle_wfa_ca_message() to be able to use a switch there - so you could just as well do the table conditions in there and forget about the enum entirely.

So something like:

```
static const struct {
    const char *command;
    std::string (*handler) (std::string command, std::vector<std::string> params);
} = {
  {"CA_GET_VERSION", [](std::string command, std::vector<std::string> params) { return "VERSION 1"; } },
...
};
```

But maybe I'm overengineering and this is simpler.

Reply:
For me this is overengineering and I would prefer to keep it as Moran did:)